### PR TITLE
fixed unininitialized value,

### PR DESCRIPTION
**Pull request by hnil**
fixed uninitialized value in the smoother args for make using ParILUO as smoother for amg. 
and added commented specialization for the solver factory

### CODE REVIEW

**Comment by GitPaean** [comment]
Should we have a parallel regression test for `flow_blackoil_dunecpr`?

**Comment by hnil** [comment]
We probably should have a test. I don not know the policy. I would have prefered one with extra tolerances of the linear solve and nonlinear solve. We should also fix a spesialization of:

File: opm/simulators/linalg/PreconditionerFactory.hpp [hunk 1]
```diff
@@ -143,12 +143,33 @@ class PreconditionerFactory
         using SmootherArgs = typename Dune::Amg::SmootherTraits<Smoother>::Arguments;
         SmootherArgs smootherArgs;
         smootherArgs.iterations = prm.get<int>("iterations", 1);
+	//int iluwitdh = prm.get<int>("iluwidth", 0);
+	//MILU_VARIANT milu = convertString2Milu(prm.get<std::string>("milutype",std::string("ilu")));
+	//smootherArgs.setMilu(milu);
+	//smootherArgs.setN(iluwitdh);
```
Comment (blattms) [inline opm/simulators/linalg/PreconditionerFactory.hpp:149]: What is up with this? Please remove commented code

File: opm/simulators/linalg/PreconditionerFactory.hpp [hunk 2]
```diff
@@ -143,12 +143,33 @@ class PreconditionerFactory
...
         // smootherArgs.overlap=SmootherArgs::vertex;
         // smootherArgs.overlap=SmootherArgs::none;
         // smootherArgs.overlap=SmootherArgs::aggregate;
-        smootherArgs.relaxationFactor = prm.get<double>("relaxation", 0.9);
+        smootherArgs.relaxationFactor = prm.get<double>("relaxation", 1.0);
```
Comment (blattms) [inline opm/simulators/linalg/PreconditionerFactory.hpp:153]: Maybe this deserves to be mentioned in the commit message.

File: opm/simulators/linalg/PreconditionerFactory.hpp [hunk 3]
```diff
@@ -143,12 +143,33 @@ class PreconditionerFactory
...
+    // 	int iluwitdh = prm.get<int>("iluwidth", 0);
+    // 	MILU_VARIANT milu = convertString2Milu( prm.get<std::string>("milutype",std::string("ilu") ) );
+    // 	smootherArgs.setMilu(milu);
+    // 	smootherArgs.setN(iluwitdh);
+    //     // smootherArgs.overlap=SmootherArgs::vertex;
+    //     // smootherArgs.overlap=SmootherArgs::none;
+    //     // smootherArgs.overlap=SmootherArgs::aggregate;
+    //     smootherArgs.relaxationFactor = prm.get<double>("relaxation", 1.0);
+    //     return smootherArgs;
+    // }
```
Comment (blattms) [inline opm/simulators/linalg/PreconditionerFactory.hpp:172]: Why the addition a a completely commented out code block?

**Comment by blattms** [comment]
Does this fix the parallel runs with cpr/flexible solvers? I had experienced the opposite  of scalability (more time using more cores) on some very easy problems.

File: opm/simulators/linalg/PreconditionerFactory.hpp (hunk 3 above)
Comment (hnil) [reply opm/simulators/linalg/PreconditionerFactory.hpp:172]: Because It should be fixed. But commenting out the code do not compile at least for me.

**Comment by hnil** [comment]
Flexible solver is just a wrapper of dune-istl solvers. So if it is problems with the scaling it need to be related to dune-istl. "Simple cases"/small cases would probably not have good scaling with cpr or rather with amg...

File: opm/simulators/linalg/PreconditionerFactory.hpp (hunk 2 above)
Comment (hnil) [reply opm/simulators/linalg/PreconditionerFactory.hpp:153]: Do you have an opinion? The default settings for flow_blackoil_dunecpr is anyway 1.0.

**Comment by hnil** [comment]
I am also unsure if the redistribution feature is compiled correctly with standard settings using OPM. It works when tested standalone.

**Comment by blattms** [comment]
> So if it is problems with the scaling it need to be related to dune-istl
????

Just been hit by this problem in my very own branch (`iluIteration_=32767`) in parallel. Maybe we can agree that even when using wrappers not all problems are upstream? (I might be to blame anyway, though ;)

Will give this a spin and merge if it fixes the problem.

**Comment by blattms** [comment]
jenkins  build this please.

**Comment by blattms** [comment]
With this change it is actually running in parallel. Now we can wonder about scaling...

@hnil Will you clean this up? I.e. remove all no-op changes. The real fix seems to be only one line.

File: opm/simulators/linalg/PreconditionerFactory.hpp (hunk 3 above)
Comment (blattms) [reply opm/simulators/linalg/PreconditionerFactory.hpp:172]: Sorry, but that is not an argument. Can't you use git stash?

**Comment by hnil** [comment]
I can clean it up, but need help with the template specialization which I commented out. It is not needed, but it will make the code more complete.

**Comment by blattms** [comment]
Well, me intend was too get the (rather urgent) bugfix merged and leave the
rest for another day. I am not a big fan of Prs including multiple things.
I might do that myself to speed things up.

**Comment by blattms** [comment]
@hnil I created PR #2614 with just the fix. The rest based on that PR can be found in [blattms/opm-simulators:hnil-complete](https://github.com/blattms/opm-simulators/tree/hnil-complete)

**Comment by atgeirr** [comment]
Replaced by #2614 and #2616.

**Pull request closed without merging**